### PR TITLE
Fixed Search Service issue for supporting GCC High tenants

### PR DESCRIPTION
**Pull request by v-smahaj**
**Issue**
Search service not working for GCC Hybrid deployments.

**Solution details** -
In GCC environment, Search service searchDnsSuffix is search.azure.us.
So added setting to decide whether deployment is GCC high or not, based on that setting appropriate searchDnsSuffix value.

**Testing Done**
Verified locally and with ARM deployment.

### CODE REVIEW

File: Source/Microsoft.Teams.Apps.RewardAndRecognition/Models/SearchServiceOptions.cs [hunk 1]
```diff
@@ -23,5 +23,10 @@ public class SearchServiceOptions
         /// Gets or sets search service admin API key.
         /// </summary>
         public string SearchServiceAdminApiKey { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether deployment type is GCC Hybrid.
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.RewardAndRecognition/Models/SearchServiceOptions.cs:28]: Please use standard terminology here

File: Source/Microsoft.Teams.Apps.RewardAndRecognition/Providers/AwardNominationSearchService.cs [hunk 2]
```diff
@@ -61,14 +61,17 @@ public class AwardNominationSearchService : IAwardNominationSearchService, IDisp
             storageOptions = storageOptions ?? throw new ArgumentNullException(nameof(storageOptions));
 
             this.searchServiceOptions = searchServiceOptions.CurrentValue;
+            var searchDnsSuffix = this.searchServiceOptions.IsGCCHybridDeployment == true ? "search.azure.us" : "search.windows.net";
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.RewardAndRecognition/Providers/AwardNominationSearchService.cs:64]: == true is redundant

File: Deployment/Botazuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
+{
```
Comment (aosolis) [inline Deployment/Botazuredeploy.json:1]: What are these 2 ARM templates, and how are they supposed to be used?

**Review by aosolis — CHANGES_REQUESTED** [review]
See comments

File: Deployment/Botazuredeploy.json (hunk 3 above)
Comment (v-smahaj) [reply Deployment/Botazuredeploy.json:1]: We are using hybrid deployment here where O365 users and Microsoft Teams service are in GCC commercial tenant and other resources are created in GCC-H subscription. So, first ARM template only creates bot services in GCC commercial subscription and second ARM template creates other resources in GCC-H subscription.

Comment (aosolis) [reply Deployment/Botazuredeploy.json:1]: I see, can you move these to a gcc subfolder then?

**Comment by aosolis** [comment]
@v-smahaj Just want to confirm-- did you test this for regular (non-GCC) deployment as well?

File: Deployment/Botazuredeploy.json (hunk 3 above)
Comment (v-smahaj) [reply Deployment/Botazuredeploy.json]: Moved GCC ARM templates to GCC subfolder.

**Comment by v-smahaj** [comment]
> @v-smahaj Just want to confirm-- did you test this for regular (non-GCC) deployment as well?

Yes. App is tested in regular tenants as well.

**Review by aosolis — APPROVED** [review]

**Pull request merged**